articles/tutorials/advanced/2d_shaders/07_sprite_vertex_effect/index.md

```diff
@@ -0,0 +1,863 @@
+
+```hlsl
+struct VSOutput
+{
+	float4 position		: SV_Position;
+	float4 color		: COLOR0;
+    float2 texCoord		: TEXCOORD0;
+};
+
+VSOutput SpriteVertexShader(	float4 position	: POSITION0,
+								float4 color	: COLOR0,
+								float2 texCoord	: TEXCOORD0)
+{
+	VSOutput output;
+    output.position = mul(position, MatrixTransform);
+	output.color = color;
+	output.texCoord = texCoord;
+	return output;
+}
+```
+
+The `SpriteVertexShader` looks different from our pixel shaders in a few important ways, 
+1. The inputs and outputs are different.
+	1. The return type is not just a `float4`, its an entire struct, `VSOutput`,
+	2. The inputs are not the same as the pixel shader. The pixel shader got a `Color` and `TextureCoordinates`, but this vertex shader has a `position`, a `color`, and a `texCoord`. 
+2. There is a `MatrixTransform` shader parameter available to this shader. 
+
+### Input Semantics
+
+The inputs to the vertex shader mirror the information that the `SpriteBatch` class bundles up for each vertex. If you look at the [`SpriteBatchItem`](xref:Microsoft.Xna.Framework.Graphics.SpriteBatchItem), you will see that each sprite is made up of 4 `VertexPositionColorTexture` instances:
+
+```csharp
+public VertexPositionColorTexture vertexTL;
+public VertexPositionColorTexture vertexTR;
+public VertexPositionColorTexture vertexBL;
+public VertexPositionColorTexture vertexBR;
+```
+
+
+The [`VertexPositionColorTexture`](xref:Microsoft.Xna.Framework.Graphics.VertexPositionColorTexture) class is a standard MonoGame implementation of the `IVertexType`, and it defines a `Position`, a `Color`, and a `TextureCoordinate` for each vertex. Those should look familiar, because they align with the inputs to the vertex shader function. The alignment is not happenstance, it is enforced by "semantics" that are applied to each field in the vertex. This snippet from the `VertexPositionColorTexture` class defines the semantics for each field in the vertex by specifying the `VertexElementUsage`:
+
+```csharp
+static VertexPositionColorTexture()
+{
+	var elements = new VertexElement[] 
+	{ 
+		new VertexElement(0, VertexElementFormat.Vector3, VertexElementUsage.Position, 0), 
+		new VertexElement(12, VertexElementFormat.Color, VertexElementUsage.Color, 0), 
+		new VertexElement(16, VertexElementFormat.Vector2, VertexElementUsage.TextureCoordinate, 0) 
+	};
+	VertexDeclaration = new VertexDeclaration(elements);
+}
+```
+
+> [!note]
+> MonoGame is open source, so you can go read the full code for [SpriteBatchItem](https://github.com/MonoGame/MonoGame/blob/develop/MonoGame.Framework/Graphics/SpriteBatchItem.cs) and[`VertexPositionColorTexture`](https://github.com/MonoGame/MonoGame/blob/develop/MonoGame.Framework/Graphics/Vertices/VertexPositionColorTexture.cs)
+
+
+The vertex shader declares a semantic for each input using the `:` syntax:
+
+```hlsl
```

Comment:
You might want to expand on this to avoid confusion, including for reference for how the implementation being demonstrated is different and Not? using SpriteBatch